### PR TITLE
GLT-1071 Improved improvement to ensure barcode uniqueness

### PR DESCRIPTION
**Pull request by tonydebat**
Checking barcode across Samples, Libraries and Pools to ensure uniqueness.

### CODE REVIEW

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 1]
```diff
@@ -395,6 +396,15 @@ public JSONObject changeLibraryIdBarcode(HttpSession session, JSONObject json) {
     try {
       User user = securityManager.getUserByLoginName(SecurityContextHolder.getContext().getAuthentication().getName());
       if (!isStringEmptyOrNull(idBarcode)) {
+        Collection<Boxable> previouslyBarcodedItems = requestManager.getBoxablesFromBarcodeList(Arrays.asList(new String[] { idBarcode }));
```
Comment (apmasell) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: `Arrays.asList` is variadic, so you can write this `Arrays.asList(idBarcode)`.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java [hunk 2]
```diff
@@ -395,6 +396,15 @@ public JSONObject changeLibraryIdBarcode(HttpSession session, JSONObject json) {
...
+        if (!previouslyBarcodedItems.isEmpty()) {
```
Comment (apmasell) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: It should probably check that the id of the current library and the retrieved library are different. `!previouslyBarcodedItems.isEmpty() && !(previouslyBarcodedItems.size() == 1 && previouslyBarcodedItems.get(0).getId() != libraryId)`, where `libraryId` is going to have to get passed in the JSON.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java (hunk 1 above)
Comment (tonydebat) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: Removed unnecessary array.

File: spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java (hunk 2 above)
Comment (tonydebat) [inline spring/src/main/java/uk/ac/bbsrc/tgac/miso/spring/ajax/LibraryControllerHelperService.java]: Added check.

**Comment by apmasell** [comment]
:+1:

**Comment by djcooke** [comment]
:+1:

**Pull request merged**